src/cli/src/root_cli.rs

```diff
@@ -150,47 +168,77 @@ enum Root {
 
     /// Print Fluvio version information
     #[structopt(name = "version")]
-    Version(VersionCmd),
+    Version(VersionOpt),
 
     /// Generate command-line completions for Fluvio
     #[structopt(
         name = "completions",
         settings = &[AppSettings::Hidden]
     )]
-    Completions(CompletionShell),
+    Completions(CompletionCmd),
 
     #[structopt(external_subcommand)]
     External(Vec<String>),
 }
 
-pub fn run_cli(args: &[String]) -> eyre::Result<String> {
-    run_block_on(async move {
-        let terminal = Arc::new(PrintTerminal::new());
-
-        let root_args: Root = Root::from_iter(args);
-        let output = match root_args {
-            Root::Consume(consume) => process_consume_log(terminal.clone(), consume).await?,
-            Root::Produce(produce) => process_produce_record(terminal.clone(), produce).await?,
-            Root::SPU(spu) => process_spu(terminal.clone(), spu).await?,
-            Root::SPUGroup(spu_group) => process_spu_group(terminal.clone(), spu_group).await?,
-            Root::CustomSPU(custom_spu) => process_custom_spu(terminal.clone(), custom_spu).await?,
-            Root::Topic(topic) => process_topic(terminal.clone(), topic).await?,
-            Root::Partition(partition) => partition.process_partition(terminal.clone()).await?,
-            Root::Profile(profile) => process_profile(terminal.clone(), profile).await?,
-            Root::Cluster(cluster) => process_cluster(terminal.clone(), cluster).await?,
-            #[cfg(any(feature = "cluster_components", feature = "cluster_components_rustls"))]
-            Root::Run(opt) => process_run(opt)?,
-            Root::Install(opt) => opt.process().await?,
-            Root::Update(opt) => opt.process().await?,
-            Root::Version(_) => process_version_cmd()?,
-            Root::Completions(shell) => process_completions_cmd(shell)?,
-            Root::External(args) => process_external_subcommand(args)?,
-        };
-        Ok(output)
-    })
-}
+impl RootCmd {
+    pub async fn process(self, root: RootOpt) -> Result<()> {
+        let out = Arc::new(PrintTerminal::new());
+        let fluvio_config = root.target.load()?;
+        let fluvio = Fluvio::connect_with_config(&fluvio_config).await?;
```

Comment:
Cluster doesn't require connect to fluvio

Reply:
Mmm, yep. This is something that I had fixed in my `zsh` branch, because I noticed that some commands don't require connecting to Fluvio at all but they were being blocked on establishing a valid connection. For example I couldn't generate completions until connecting to a cluster. I have a fix for this, let me bring it in here.